python/cudf_polars/cudf_polars/experimental/dask_serialize.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Dask serialization."""
+
+from __future__ import annotations
+
+import pylibcudf as plc
+
+from cudf_polars.containers import DataFrame
+
+try:
+    import cupy
+    from distributed.protocol import (
+        dask_deserialize,
+        dask_serialize,
+    )
+    from distributed.protocol.cuda import (
+        cuda_deserialize,
+        cuda_serialize,
+    )
+    from distributed.utils import log_errors
+
+    @cuda_serialize.register(DataFrame)
+    def _(x):
+        with log_errors():
+            return x.serialize()
+
+    @cuda_deserialize.register(DataFrame)
+    def _(header, frames):
+        with log_errors():
+            return DataFrame.deserialize(header, frames)
+
+    @dask_serialize.register(DataFrame)
+    def _(x):
+        with log_errors():
+            header, frames = x.serialize()
+            # Copy GPU buffers to host and record it in the header
+            gpu_frames = [
+                i
+                for i in range(len(frames))
+                if isinstance(frames[i], plc.gpumemoryview)
+            ]
+            for i in gpu_frames:
+                frames[i] = memoryview(cupy.asnumpy(frames[i]))
+            header["gpu_frames"] = gpu_frames
+            return header, frames
```

Comment:
Since we know serialize always returns `header, (hostview, deviceview)` this can be simpler, I think:

```
header, (metadata, data_d) = x.serialize()
return header, (metadata, memoryview(cupy.asnumpy(data_d)))
```

Related, this seems like it would make sense for `gpumemoryview` to implement `from_memoryview` and `to_memoryview` methods. Then we don't need to use cupy for a very heavyweight `cudamemcpy`.

Reply:
Or at least, use RMM device buffers:

```
# from host
plc.gpumemoryview(rmm.DeviceBuffer.to_device(data_h))

# to host
memoryview(rmm.DeviceBuffer(ptr=data_d.ptr, size=data_d.__cuda_array_interface__["shape"][0]).copy_to_host())
```

This way, we also don't have to worry about configuring cupy to use the same memory resource we're using inside cudf-polars.